### PR TITLE
Add a local client that uses RPC but does not use a network connection

### PR DESCRIPTION
**Pull request by bartv**
# Description

This PR does a number of things

- Introduce a local client that does not use a network connection.
- Introduce a typed client that already parses some of the json and restores exceptions
- Move protocol tests in the protocol folder

# Self Check:

Strike through any lines that are not applicable (`~~line~~`) then check the box

- [ ] Attached issue to pull request
- [ ] Changelog entry
- [ ] Type annotations are present
- [ ] Code is clear and sufficiently documented
- [ ] No (preventable) type errors (check using make mypy or make mypy-diff)
- [ ] Sufficient test cases (reproduces the bug/tests the requested feature)
- [ ] Correct, in line with design
- [ ] End user documentation is included or an issue is created for end-user documentation (add ref to issue here: )
- [ ] If this PR fixes a race condition in the test suite, also push the fix to the relevant stable branche(s) (see [test-fixes](https://internal.inmanta.com/development/core/tasks/build-master.html#test-fixes) for more info)

### CODE REVIEW

**Review by wouterdb — APPROVED** [review]

File: src/inmanta/protocol/endpoints.py [hunk 1]
```diff
@@ -474,3 +477,65 @@ async def _call(
...
+            case _:
+                self._raise_exception(exceptions.ServerError, response.result)
+
+    async def _call(
+        self, method_properties: common.MethodProperties, args: list[object], kwargs: dict[str, object]
+    ) -> types.ReturnTypes:
+        """Execute a call and return the result"""
+        return self._process_response(method_properties, await self._transport_instance.call(method_properties, args, kwargs))
+
+    def __getattr__(self, name: str) -> Callable[..., Coroutine[Any, Any, types.ReturnTypes]]:
```
Comment (wouterdb) [inline src/inmanta/protocol/endpoints.py:539]: why is this here? (It looks like the default behavior of python re-defined)

**Review by wouterdb — APPROVED** [review]

File: src/inmanta/protocol/endpoints.py (hunk 1 above)
Comment (bartv) [reply src/inmanta/protocol/endpoints.py]: I removed it

File: tests/protocol/test_typed_client.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+from inmanta import protocol
+from inmanta.data.model import BaseModel
+from inmanta.postgresproc import PostgresProc
+from inmanta.protocol import exceptions
+from inmanta.server.protocol import LocalClient, Server, ServerSlice, common
+from utils import configure
+
+
+async def test_local_client(unused_tcp_port: int, postgres_db: PostgresProc, database_name: str, async_finalizer) -> None:
+    """The the local clien"""
```
Comment (arnaudsjs) [inline tests/protocol/test_typed_client.py:32]:
```suggestion
    """Test the local client"""
```

**Review by arnaudsjs — APPROVED** [review]
